src/swarm/neo/request/RequestEventDispatcher.d

```diff
@@ -629,39 +634,56 @@ public struct RequestEventDispatcher
 
     private void register ( MessageFiber fiber, EventRegistration event )
     {
-        with ( event.Active ) switch ( event.active )
+        // See if a matching element is already in this.waiting_fibers.
+        WaitingFiber* in_list;
+        foreach ( ref waiting_fiber; this.waiting_fibers )
         {
-            case message:
-                foreach ( waiting_fiber; this.waiting_fibers )
-                    if ( waiting_fiber.event.active == message )
+            if ( waiting_fiber.fiber == fiber && waiting_fiber.event == event )
+            {
+                assert(in_list is null);
+                assert(!waiting_fiber.enabled);
+                in_list = &waiting_fiber;
+            }
+
+            // Ensure the user isn't trying to do anything crazy.
+            if ( waiting_fiber.enabled &&
+                waiting_fiber.event.active == event.active )
+            {
+                with ( event.Active ) switch ( event.active )
+                {
+                    case message:
                         enforce(waiting_fiber.event.message.type
```

Comment:
This is something else, not related to this commit, but shouldn't this be `verify` ?

Reply:
I suppose you're right. It would be a programmer error, not an error based on processing run-time data. `verify` doesn't exist in the ocean version that swarm uses, but these could be turned into `assert`s.

Reply:
Maybe just add a comment here, so we know when we `verify` everything?

Reply:
Ping here, and LGTM.

Reply:
Two things: 

- `enforce`` <- two ticks at the end
- maybe add `should be asserts (and ideally verify)` so we can `grep` for them when do the transition.

Reply:
Ah yes. Updated.

Reply:
Oh yeah. Updated. (The comments already contain the string "assert".)

Reply:
Yes, I just saw it, sorry!